stellar_rust_sdk/src/trades/all_trades_request.rs

```diff
@@ -0,0 +1,179 @@
+    ///
+    /// # Returns
+    ///
+    /// The updated `AllTradesRequest` with the base asset set.    
+    pub fn set_base_asset(
+        self,
+        base_asset: AssetType,
+    ) -> Result<AllTradesRequest, String> {
+        Ok(AllTradesRequest {
+            base_asset: Some(BaseAsset(base_asset)),
+            ..self
+        })
+    }
+
+    /// Specifies the counter asset in the request.
+    ///
+    /// # Arguments
+    ///
+    /// * `counter_asset` - The counter asset type to filter the trades. It can be one of the following:
+    ///     - `AssetType::Native`
+    ///     - `AssetType::Alphanumeric4(AssetData)`
+    ///     - `AssetType::Alphanumeric12(AssetData)`
+    ///
+    /// # Returns
+    ///
+    /// The updated `AllTradesRequest` with the counter asset set.    
+    pub fn set_counter_asset(
+        self,
+        counter_asset: AssetType,
+    ) -> Result<AllTradesRequest, String> {
+        Ok(AllTradesRequest {
+            counter_asset: Some(CounterAsset(counter_asset)),
+            ..self
+        })
+    }
+}
+
+impl Request for AllTradesRequest {
+    fn get_query_parameters(&self) -> String {
+        let mut query: Vec<String> = Vec::new();
+        
+        if let Some(base_asset) = &self.base_asset {
+            match &base_asset.0 {
+                AssetType::Native => {
+                    query.push(format!("base_asset_type=native"));
+                }
+                AssetType::Alphanumeric4(asset) => {
+                    query.push(format!("base_asset_type=credit_alphanum4"));
+                    query.push(format!("&base_asset_code={}", asset.asset_code));
+                    query.push(format!("&base_asset_issuer={}", asset.asset_issuer));
+                }
+                AssetType::Alphanumeric12(asset) => {
+                    query.push(format!("base_asset_type=credit_alphanum12"));
+                    query.push(format!("&base_asset_code={}", asset.asset_code));
+                    query.push(format!("&base_asset_issuer={}", asset.asset_issuer));
+                }
+            }
+        }
+
+        if let Some(base_asset) = &self.base_asset {
+            match &base_asset.0 {
+                AssetType::Native => {
+                    query.push(format!("&counter_asset_type=native"));
+                }
+                AssetType::Alphanumeric4(asset) => {
+                    query.push(format!("&counter_asset_type=credit_alphanum4"));
+                    query.push(format!("&counter_asset_code={}", asset.asset_code));
+                    query.push(format!("&counter_asset_issuer={}", asset.asset_issuer));
+                }
+                AssetType::Alphanumeric12(asset) => {
+                    query.push(format!("&counter_asset_type=credit_alphanum12"));
+                    query.push(format!("&counter_asset_code={}", asset.asset_code));
+                    query.push(format!("&counter_asset_issuer={}", asset.asset_issuer));
+                }
+            }
```

Comment:
Is deze code nu duplicaat? De check lijkt hetzelfde, maar de implementatie verschilt licht. Wellicht bedoelde je bij de 2e check geen `base_asset` maar een `counter_asset`?

Reply:
Ah, ja, klopt helemaal wat je zegt.